clippy_lints/src/path_buf_push_overwrite.rs

```diff
@@ -0,0 +1,65 @@
+use crate::utils::{match_type, paths, span_help_and_lint, walk_ptrs_ty};
+use if_chain::if_chain;
+use rustc::hir::*;
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::{declare_tool_lint, lint_array};
+use std::path::Path;
+use syntax::ast::LitKind;
+
+declare_clippy_lint! {
+    /// **What it does:*** Checks for `push` calls on `PathBuf` that can cause overwrites.
+    ///
+    /// **Why is this bad?** Calling `push` with a root path at the start can overwrite the
+    /// previous defined path.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    /// ```ignore
```

Comment:
You can change this from `ignore` to `rust` by adding
```suggestion
    /// ```rust
    /// # use std::path::PathBuf;
```

Same below